In [50]:
import pandas as pd
import psycopg2

In [51]:
titanic = pd.read_csv('https://raw.githubusercontent.com/lucchesia7/coding-temple-da-sql-day-1/main/day_1/data/titanic.csv')
titanic.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses_Aboard  Parents/Children_Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [52]:
titanic.columns = titanic.columns.str.lower().str.replace(' ', '_').str.replace('/', '_')

In [53]:
display(titanic)

survived  pclass                                               name  \
0           0       3                             Mr. Owen Harris Braund   
1           1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2           1       3                              Miss. Laina Heikkinen   
3           1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4           0       3                            Mr. William Henry Allen   
..        ...     ...                                                ...   
882         0       2                               Rev. Juozas Montvila   
883         1       1                        Miss. Margaret Edith Graham   
884         0       3                     Miss. Catherine Helen Johnston   
885         1       1                               Mr. Karl Howell Behr   
886         0       3                                 Mr. Patrick Dooley   

        sex   age  siblings_spouses_aboard  parents_children_aboard     fare  
0      male  22.0                        1                        0   7.2500  
1    female  38.0                        1                        0  71.2833  
2    female  26.0                        0                        0   7.9250  
3    female  35.0                        1                        0  53.1000  
4      male  35.0                        0                        0   8.0500  
..      ...   ...                      ...                      ...      ...  
882    male  27.0                        0                        0  13.0000  
883  female  19.0                        0                        0  30.0000  
884  female   7.0                        1                        2  23.4500  
885    male  26.0                        0                        0  30.0000  
886    male  32.0                        0                        0   7.7500  

[887 rows x 8 columns]

In [58]:
connection ='postgresql://dfhnuoff:xE2zkIwMNSCoT0qmJ8DJRJXPppXSPEQm@peanut.db.elephantsql.com/dfhnuoff'

In [59]:
titanic.to_sql('titanic', con = connection)

887

Question 1: How many rows of data are there?

```sql

SELECT COUNT(*) FROM titanic

```

Question 1 Answer: 887

Question 2: How many people survived?

```sql

SELECT SUM(survived) FROM titanic

```

Question 2 Answer: 342

Question 3:  What passenger class has the largest population?

```sql

SELECT pclass, COUNT(pclass) AS pclass_count FROM titanic
GROUP BY pclass
ORDER by COUNT(pclass)
DESC

```

Question 3 Answer: Pclass 3 had 487 passengers in it

*Bonus Question: List all passengers that their name starts with Miss

```sql 

SELECT name
FROM titanic
WHERE name LIKE 'Miss%';


```

*Bonus Answer:

Miss. Laina Heikkinen
Miss. Marguerite Rut Sandstrom
Miss. Elizabeth Bonnell
Miss. Hulda Amanda Adolfina Vestrom
Miss. Anna McGowan
Miss. Torborg Danira Palsson
Miss. Ellen O'Dwyer
Miss. Mary Agatha Glynn
Miss. Augusta Maria Vander Planke
Miss. Jamila Nicola-Yarred
Miss. Simonne Marie Anne Andree Laroche
Miss. Margaret Delia Devaney
Miss. Bridget O'Driscoll
Miss. Emily Rugg
Miss. Constance Mirium West
Miss. Amelie Icard
Miss. Erna Alexandra Andersson
Miss. Lillian Amy Goodwin
Miss. Elizabeth Dowdell
Miss. Brigdet Delia McDermott
Miss. Bertha Ilett
Miss. Mabel Helen Fortune
Miss. Matilda Petranec
Miss. Anna Kristine Salkjelsvik
Miss. Bertha Moran
Miss. Hileni Zabour
Miss. Katriina Jussila
Miss. Malake Attalah
Miss. Ellis Anna Maria Andersson
Miss. Susan Webber
Miss. Anna Peter
Miss. Helen Monypeny Newsom
Miss. Anna Sofia Nysten
Miss. Robina Maggie Ford
Miss. Katherine Gilnagh
Miss. Eleanor Ileen Johnson
Miss. Ann Elizabeth Isham
Miss. Constance Gladys Sage
Miss. Luise Gretchen Kink-Heilmann
Miss. Carla Christine Nielsine Andersen-Jensen
Miss. Elise Lurette
Miss. Margaret Madigan
Miss. Henriette Yrois
Miss. Telma Matilda Strom
Miss. Helen Carr
Miss. Clear Annie Cameron
Miss. Madeleine Newell
Miss. Eliina Honkanen
Miss. Albina Bazzani
Miss. Mathilde Lefebre
Miss. Lillian Gertrud Asplund
Miss. Alice Phoebe Harknett
Miss. Marjorie Collyer
Miss. Thamine Zabour
Miss. Katherine Murphy
Miss. Agda Thorilda Viktoria Lindahl
Miss. Gladys Cherry
Miss. Anna Ward
Miss. Delia Henry
Miss. Amelia Bissette
Miss. Hanora Healy
Miss. Kornelia Theodosia Andrews
Miss. Augusta Charlotta Lindblom
Miss. Kate Connolly
Miss. Ellen Barber
Miss. Aloisia Haas
Miss. Helen Loraine Allison
Miss. Anna Katherine Kelly
Miss. Nora A Keane
Miss. Margaret Fleming
Miss. Laura Mabel Francatelli
Miss. Margaret Bechstein Hays
Miss. Emily Borie Ryerson
Miss. Helmina Josefina Nilsson
Miss. Mary Natalie Wick
Miss. Hilda Mary Slayter
Miss. Marie Grice Young
Miss. Jean Gertrude Hippach
Miss. Agnes McCoy
Miss. Elizabeth Margaret Burns
Miss. Alice Elizabeth Fortune
Miss. Amelia Brown
Miss. Marion Elsie Smith
Miss. Elsie Edith Bowerman
Miss. Annie Clemmer Funk
Miss. Mary McGovern
Miss. Helen Mary Mockler
Miss. Annie Jermyn
Miss. Stina Viola Palsson
Miss. Aurora Adelia Landergren
Miss. Rosalie Bidois
Miss. Maria Nakid
Miss. Kate Buss
Miss. Bertha Lehmann
Miss. Marjorie Newell
Miss. Elina Olsson
Miss. Mari Aina Jussila
Miss. Marija Oreskovic
Miss. Ida Lefebre
Miss. Daisy E Minahan
Miss. Lyyli Karoliina Silven
Miss. Catharina Van Impe
Miss. Kate Florence Phillips
Miss. Lucile Polk Carter
Miss. Doolina Margaret Ford
Miss. Madeleine Violet Mellinger
Miss. Marie Catherine Baclini
Miss. Ellen Toomey
Miss. Helene Barbara Baclini
Miss. Ida Sofia Strandberg
Miss. Hildur E Hirvonen
Miss. Jeannie Lefebre
Miss. Elizabeth Mussey Eustis
Miss. Mary Canavan
Miss. Bridget Mary O'Sullivan
Miss. Kristina Sofia Laitinen
Miss. Roberta Maioni
Miss. Anne Perreault
Miss. Lucy Ridsdale
Miss. Phyllis May Quick
Miss. Marija Cacic
Miss. Eva Miriam Hart
Miss. Bertha LeRoy
Miss. Hedwig Margaritha Frolicher
Miss. Harriet R Crosby
Miss. Ingeborg Constanzia Andersson
Miss. Sigrid Elisabeth Andersson
Miss. Velin Ohman
Miss. (Marion Ogden) Meanwell
Miss. Mary Kelly
Miss. Ethel Garside
Miss. Julie Rachel Christy
Miss. Ruth Taussig
Miss. Mary Bourke
Miss. Jessie Wills Leitch
Miss. Elizabeth W Shutes
Miss. Margaret Jane Murphy
Miss. Alice Herman
Miss. Marion Louise Becker
Miss. Gretchen Fiske Longley
Miss. Mabel Skoog
Miss. Mary Davis
Miss. Margit Elizabeth Skoog
Miss. Eugenie Baclini
Miss. Amy Zillah Elsie Stanley
Miss. Elsie Doling
Miss. Hanora O'Leary
Miss. Hanora Hegarty
Miss. Anna Sofia Turja
Miss. Katie Peters
Miss. Georgette Alexandra Madill
Miss. Manca Karun
Miss. Katherine Mullens
Miss. Saiide Barbara
Miss. Alice Cleaver
Miss. Caroline Louise Endres
Miss. Edwina Celia Troutt
Miss. Annie Jessie Harper
Miss. Margareth Mannion
Miss. Pieta Sofia Ilmakangas
Miss. Elisabeth Walton Allen
Miss. Susan Parker Ryerson
Miss. Anna Sinkkonen
Miss. Joan Wells
Miss. Mary Mangan
Miss. Virginia Ethel Emanuel
Miss. Banoura Ayoub
Miss. Anna Sofia Sjoblom
Miss. Stella Anna Sage
Miss. Ellen Natalia Pettersson
Miss. Ebba Iris Alfrida Andersson
Miss. Wendla Maria Heininen
Miss. Sara Rebecca Compton
Miss. Augusta Serepeca
Miss. Nourelain Boulos
Miss. Mary Conover Lines
Miss. Dorothy Edith Sage
Miss. Asuncion Duran y More
Miss. Adele Kiamie Najib
Miss. Gerda Ulrika Dahlberg
Miss. Margaret Edith Graham
Miss. Catherine Helen Johnston